app/uk/gov/hmrc/eoricommoncomponent/frontend/controllers/EmailController.scala

```diff
@@ -51,53 +54,55 @@ class EmailController @Inject() (
   appConfig: AppConfig,
   enrolmentPendingForUser: enrolment_pending_for_user,
   enrolmentPendingAgainstGroupId: enrolment_pending_against_group_id,
-  emailJourneyService: EmailJourneyService
+  emailJourneyService: EmailJourneyService,
+  errorPage: error_template
 )(implicit ec: ExecutionContext)
     extends CdsController(mcc) with EnrolmentExtractor {
 
   def form(service: Service): Action[AnyContent] =
     authAction.ggAuthorisedUserWithEnrolmentsAction { implicit request => implicit user: LoggedInUserWithEnrolments =>
-      startRegisterJourney(service)
+      startRegisterJourney(service).foldF(_ => Future.successful(InternalServerError(errorPage(service))), res => res)
     }
 
-  private def startRegisterJourney(
-    service: Service
-  )(implicit hc: HeaderCarrier, request: Request[AnyContent], user: LoggedInUserWithEnrolments) =
-    groupEnrolment.groupIdEnrolments(user.groupId.getOrElse(throw MissingGroupId())).flatMap {
-      groupEnrolments =>
-        if (groupEnrolments.exists(_.service == service.enrolmentKey))
-          // user has specified service
-          if (service.code.equalsIgnoreCase(appConfig.standaloneServiceCode))
-            existingEoriForUserOrGroup(user, groupEnrolments) match {
+  private def startRegisterJourney(service: Service)(implicit
+    hc: HeaderCarrier,
+    request: Request[AnyContent],
+    user: LoggedInUserWithEnrolments
+  ): EitherT[Future, ResponseError, Future[Result]] =
+    groupEnrolment.groupIdEnrolments(user.groupId.getOrElse(throw MissingGroupId()))
+      .map(
```

Comment:
We can either flatmap here and the below can return an either, or we can fold here returning the InternalServerError here for a left instead of in the form method.